invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/src/deposit/fields/UppyUploader/InvenioMultipartUploader.js

```diff
@@ -0,0 +1,384 @@
+   *   - **method**: string, the HTTP method to be used for the upload. This should be one of either PUT or POST, depending on the type of upload used.
+   *   - **url**: string, the URL to which the upload request will be sent. When using a presigned PUT upload, this should be the URL to the S3 object with signing parameters included in the query string. When using a POST upload with a policy document, this should be the root URL of the bucket.
+   *   - **fields**: object, an object with form fields to send along with the upload request. For presigned PUT uploads (which are default), this should be left empty.
+   *   - **headers**: object, an object with request headers to send along with the upload request. When using a presigned PUT upload, it’s a good idea to provide headers['content-type']. That will make sure that the request uses the same content-type that was used to generate the signature. Without it, the browser may decide on a different content-type instead, causing S3 to reject the upload.
+   */
+  async getUploadParameters(file, options, signal) {
+    file.transferOptions = {
+      fileSize: file.size,
+      type: "L",
+      cancelFn: () => this.uppy.removeFile(file.id),
+    };
+
+    try {
+      const response = await this.opts.initializeFileUpload(this.draftRecord, file);
+      this.uppy.setFileMeta(file.id, {
+        file_id: response.file_id,
+        links: response.links,
+      });
+      return {
+        headers: {
+          "Content-Type": "application/octet-stream",
+        },
+        url: response.links.content,
+        method: "PUT",
+        signal,
+      };
+    } catch (error) {
+      this.uppy.info(error.message, "error");
+      throw error;
+    }
+  }
+
+  /**
+   * A function that returns the minimum chunk size to use when uploading the given file as multipart.
+   * For multipart uploads, chunks are sent in batches to have presigned URLs generated with signPart(). To reduce the amount of requests for large files, you can choose a larger chunk size, at the cost of having to re-upload more data if one chunk fails to upload.
+   * S3 requires a minimum chunk size of 5MiB, and supports at most 10,000 chunks per multipart upload. If getChunkSize() returns a size that’s too small, Uppy will increase it to S3’s minimum requirements.
+   * Default implementation produces a chunk size between 5-250 MB for max. 10240 chunks (max filesize. ~2.44 TiB).
+   * @param {*} file the file object from Uppy’s state. The most relevant keys are file.name and file.type
+   * @returns byte size of individual part chunk
+   */
+  getChunkSize(file) {
+    const MiB = 1024 * 1024;
+    const minPartSize = MiB * 5;
+    const midPartSize = MiB * 25;
+    const maxPartSize = MiB * 250;
+
+    const smallFile = file.size <= this.#maxMultipartParts * minPartSize;
+    const mediumFile = file.size <= this.#maxMultipartParts * midPartSize;
+    const largeFile = file.size <= this.#maxMultipartParts * maxPartSize;
```

Comment:
how does it correspond to our config variable `RDM_FILES_DEFAULT_MAX_FILE_SIZE`? If uppy is configured with max file size of 2TiB but invenio has max 250mb, what will happen?

Reply:
In my tests I just had to change `FILES_REST_DEFAULT_QUOTA_SIZE`. But perhaps my tested file sizes (up to 40 GB) just were not big enough.

Reply:
I'm a little bit confused by the doc line
`Default implementation produces a chunk size between 5-250 MB for max. 10240 chunks (max filesize. ~2.44 TiB).` Why is it here 10240, the S3 docs say it's max 10000 chunks?! 
Personally I would prefer supporting up to 500 MiB per chunk, so it's nearer to the theoretical limit.

Reply:
You're right, I'll fix the docstring, it should really bé 10k

Reply:
Please see here: https://github.com/inveniosoftware/invenio-app-rdm/pull/2994/files#diff-a779205cb118ac986e91089f0dcc0ed648e4b7a32b252151438c76a8fab7f10fR366 and here: https://github.com/inveniosoftware/invenio-rdm-records/pull/2065/files#diff-ba00af4448c63206ce501ca177495455066afa22149462ca9aa01d71b3e805bcR263-R301 . It now calculates 5 log-scaled steps from 5 MiB up to the max file size limit given by `RDM_FILES_DEFAULT_MAX_FILE_SIZE`, and choses the biggest possible chunk size for a file.